src/tactic/unfold_cases.lean

```diff
@@ -0,0 +1,72 @@
+/-
+Copyright (c) 2020 Dany Fabian. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Dany Fabian
+-/
+
+import tactic.split_ifs
+namespace tactic
+
+open expr
+
+namespace unfold_cases
+meta def find_splitting_expr : expr → tactic expr
+| `(@ite %%cond %%dec_inst _ _ _ = _) := pure `(@decidable.em %%cond %%dec_inst)
+| `(%%l@(app x y) = _) := pure y
+| e := fail!"expected an expression of the form: f x = y. Got:\n{e}"
+
+meta def unfold_cases_core (inner : interactive.itactic) : tactic unit :=
+inner <|>
+(do split_ifs [], all_goals unfold_cases_core, skip) <|>
+do
+  tgt ← target,
+  e ← find_splitting_expr tgt,
+  focus1 $ do
+    cases e,
+    all_goals $ (dsimp_target >> unfold_cases_core) <|> skip,
+    skip
+
+meta def unfold_tgt : expr → tactic unit
```

Comment:
Add header comment to describe the kind of goal it operates on and what the outcome is